Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogueBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly DateTime _utcNow = DateTime.UtcNow;
+
+        private static readonly Emoji _approveEmoji = new Emoji("👍");
+
+        private static readonly Emoji _disproveEmoji = new Emoji("👎");
+
+        private IDesignatedChannelService _designatedChannelService { get; }
+
+        private IDiscordSocketClient _discordSocketClient { get; }
+
+        private IMemoryCache _memoryCache { get; }
+
+        private IPromotionsService _promotionsService { get; }
+
+        private IImageService _imageService { get; }
+
+        public PromotionDialogueBehavior(IDesignatedChannelService designatedChannelService,
+            IDiscordSocketClient discordSocketClient,
+            IMemoryCache memoryCache,
+            IPromotionsService promotionsService,
+            IImageService imageService)
+
+        {
+            _designatedChannelService = designatedChannelService;
+            _discordSocketClient = discordSocketClient;
+            _memoryCache = memoryCache;
+            _promotionsService = promotionsService;
+            _imageService = imageService;
+        }
+
+        public async Task HandleNotificationAsync(PromotionActionCreatedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return;
+
+            switch (notification.Data.Type)
+            {
+                case PromotionActionType.CampaignCreated:
+                    await CreatePromoPoll(notification.Data);
+                    break;
+                case PromotionActionType.CampaignClosed:
+                    await DeletePromoPoll(notification.Data);
+                    break;
+            }
+        }
+
+        public async Task HandleNotificationAsync(ReactionAddedNotification notification,
+            CancellationToken cancellationToken = default)
+        {
+            var reaction = notification.Reaction.Emote;
+            var user = (IGuildUser)notification.Reaction.User.Value;
+
+            if (cancellationToken.IsCancellationRequested
+                || notification.Reaction.User.Value.IsBot
+                || notification.Message.Value.Embeds.Count < 1)
+                return;
+
+            if (_memoryCache.TryGetValue(notification.Message.Id, out CachedPromoPoll poll)
+            && ValidateVoteReaction(reaction))
+            {
+                await notification.Message.Value.RemoveReactionAsync(notification.Reaction.Emote, notification.Reaction.User.Value);
```

Comment:
Just to highlight this point and assuage a common fear as it has come up several times, the reaction is removed nearly instantly before any other logic or query is done. This makes the voting anonymous